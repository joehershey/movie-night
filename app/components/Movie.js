import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  View,
  Text,
  TouchableWithoutFeedback,
  TextInput,
} from "react-native";
import { Card } from "react-native-elements";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

/* This component returns one Card representing a movie searched */
function Movie(props) {
  const [movieId, setMovieId] = useState("");
  const [movie, setMovie] = useState("");
  const [added, setAdded] = useState(false);
  const [expanded, setExpanded] = useState(false); //used when editing alias

  function addMovieToQueueAPI() {
    /* console.log(props.url);
    console.log(props.group_id);
    console.log(props.movie.id);
    console.log(props.user_id);
    fetch(props.url + "group/" + props.group_id + "/movie", {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json; charset=utf-8",
      },
      method: "POST",
      body: JSON.stringify({
        tmdb_movie_id: props.movie.id,
        added_by: props.user_id,
      }),
    })
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.group_id === undefined) {
          alert("Failed to add movie:" + responseJson.message);
        } else {
          console.log(responseJson);
          setAdded(true);
        }
      })
      .catch((error) => {
        console.error(error);
      }); */

    setAdded(true);
  }
  return (
    <View>
      <Card containerStyle={{ padding: 0, borderRadius: 10 }}>
        <View
          style={{
            flexDirection: "row",
            alignItems: "center",
          }}
        >
          <View
            style={{
              flex: 1,
              backgroundColor: "white",
              margin: 10,
              alignItems: "center",
            }}
          ></View>

          <View
            style={{
              flex: 10,
              padding: 15,
              flexDirection: "row",
              justifyContent: "center",
              alignItems: "center",
            }}
          >
            <Text style={{ fontWeight: "bold", textAlign: "center" }}>
              {props.movie.title}{" "}
              {props.movie.release_date
                ? "(" + props.movie.release_date.substring(0, 4) + ")"
                : ""}
            </Text>
          </View>
          <TouchableWithoutFeedback
            onPress={() => {
              setExpanded(!expanded);
            }}
          >
            <View
              style={{
                flex: 1,
                backgroundColor: "white",
                margin: 10,
                alignItems: "center",
              }}
            >
              <FontAwesome5
                name={expanded ? "compress" : "edit"}
                color="grey"
                size={20}
              />
            </View>
          </TouchableWithoutFeedback>
        </View>
        {expanded && <Card.Divider />}

        {expanded && (
          <View>
            <View style={{ padding: 15 }}>
              <Text>{props.movie.overview}</Text>
            </View>
            <View style={{ flexDirection: "row", margin: 0 }}>
              <TouchableWithoutFeedback onPress={() => addMovieToQueueAPI()}>
                <View
                  style={{
                    margin: 10,
                    marginTop: 0,
                    height: 50,
                    borderRadius: 5,
                    flex: 1,
                    padding: 5,
                    backgroundColor: added ? "#77a2c9" : COLORS.secondary,
                    alignItems: "center",
                    justifyContent: "center",
                  }}
                >
                  {added ? (
                    <View
                      style={{ flexDirection: "row", alignItems: "center" }}
                    >
                      <Text
                        style={{
                          color: "white",
                          fontSize: 14,
                          alignContent: "center",
                          margin: 5,
                        }}
                      >
                        Added
                      </Text>
                      <FontAwesome5 color="white" size={20} name="check" />
                    </View>
                  ) : (
                    <Text
                      style={{
                        color: "white",
                        fontSize: 14,
                        alignContent: "center",
                      }}
                    >
                      Add
                    </Text>
                  )}
                </View>
              </TouchableWithoutFeedback>
            </View>
          </View>
        )}
      </Card>
    </View>
  );
}

export default Movie;
