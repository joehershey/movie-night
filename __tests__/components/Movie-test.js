import Movie from "../../app/components/Movie";
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test Movie', () => {
    beforeEach( async () => {
        props = {
            movie: {
                poster_path: "defined"
            }
        }
        global.tree = renderer.create(<Movie {...props} />);
    });

    it("Movie renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });
});