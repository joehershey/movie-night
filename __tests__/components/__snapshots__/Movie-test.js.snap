// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Movie Movie renders properly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 0,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 3,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w500defined",
              }
            }
            style={
              Object {
                "height": 90,
                "margin": 10,
                "width": 60,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 8,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
             
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flex": 2,
              "margin": 10,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
